### PR TITLE
installation and uninstallation command added from agent installer algo

### LINKED ISSUE #486: Create installation / uninstallation scripts from agent installer algo
Create `installation` / `uninstallation` scripts from agent installer [algo](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/agent/installer/internal/algo/ubuntu20_4K8s1_22.go).    
We will be storing this algo as bash script and the new installer controller in #467 will be using these scripts to create the installation secret.

### PR DESCRIPTION
**Pull request by mayur-tolexo**
**What this PR does / why we need it**:

Added the consolidated steps that need to perform for installing/uninstalling of supported k8s version on supported os as variables in installer package

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #486

### CODE REVIEW

File: agent/installer/steps.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+// contains the installation and uninstallation steps for the supported os and k8s
+var (
+	DoUbuntu20_4K8s1_22 = `
+set -euo pipefail
+
+BUNDLE_PATH=${BUNDLE_PATH:-"/var/lib/byoh/bundles"}
+
+## disable swap
+swapoff -a && sed -ri '/\sswap\s/s/^#?/#/' /etc/fstab
```
Comment (shamsher31) [inline agent/installer/steps.go:14]: I was assuming these scripts will be generated by the agent installer and then assigned to `DoUbuntu20_4K8s1_22 ` variable.

Aren't we duplicating these scripts, one in the agent installer and the other in these variables?

Comment (mayur-tolexo) [reply agent/installer/steps.go:14]: Agreed, for some time these script will be duplicated.

we are planing to update the byo agent to install the bundle two ways:
1. agent will install the way its currently doing
2. new `K8sInstallationConfig` controller will generate the installation script and agent will execute it

These variables will be used by new `K8sInstallationConfig` controller to create the secret.

File: agent/installer/steps.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package installer
+
+// contains the installation and uninstallation steps for the supported os and k8s
+var (
+	DoUbuntu20_4K8s1_22 = `
+set -euo pipefail
```
Comment (shamsher31) [inline agent/installer/steps.go:9]: Do we also need to set `#!/usr/bin/env bash` at the top of the script as it will run as a bash script?

Comment (mayur-tolexo) [reply agent/installer/steps.go:9]: we will be executing this script using `cmd := exec.Command("/bin/bash", DoUbuntu20_4K8s1_22)` so `#!/usr/bin/env bash` won't be  required here.

**Review by shamsher31 — APPROVED** [review]
lgtm

**Review by anusha94 — APPROVED** [review]
lgtm

**Comment by anusha94** [comment]
We can also consider a different location for this file. Eventually, we want to move away from having much code in the byoh binary.
So maybe in a future PR.

**Pull request merged**